### ISSUE TITLE
In tN, Code appearing in selection pane when there is no highlight in the scripture pane

### ISSUE THREAD

**Issue by elsylambert** [unfoldingWord/translationCore#6192]
# 1.2.0 (f215a2f) 
Here is the Project: 
[en_ust_rut_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3340122/en_ust_rut_book.zip)

![object.jpg](https://images.zenhubusercontent.com/5d02efb2f425650e637abe10/7b930fcb-eff7-40fb-a0b9-65a1d87a56f9)

In tN, with the attached project, Ruth 1:16, 2:9, 2:18 and 4:12, shows up code in the "Please select translation for" pane

**Comment by PhotoNomad0** [comment]
@elsylambert @cckozie - Interesting challenge here: 
- in the uhb we have multiple words connected with zero-space joiners marked as a word, but technically they are multiple words.  
- Looks like we need to update the source content updater to leave them connected when parsing the quote.  Will double check the helps content

**Comment by PhotoNomad0** [comment]
@elsylambert - please add the category as well as the scripture reference since a verse can have multiple checks in it.

**Comment by PhotoNomad0** [comment]
@elsylambert - the fix for this issue should also fix the problem with missing highlights in the uhb and ult for these verses.

**Comment by elsylambert** [comment]
@PhotoNomad0 - Here are the categories and the references:
Ruth 1:16 - Assumed Knowledge and Implicit Information
Ruth 2:9 - Metonymy
Ruth 2:18 - Assumed Knowledge and Implicit Information
Ruth 4:12 - Metaphor

**Comment by PhotoNomad0** [comment]
Verified in translationCore 1.2.0 (b219729)

**Comment by elsylambert** [comment]
Verified in 1.2.0 (a3ee18b), cannot find the issue with the project.

@PhotoNomad0  However, I imported a new project from door43: 
[en_ult_rut_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3369269/en_ult_rut_book.zip)

Please note that under Category> Assumed Knowledge and Implicit Information> Ruth 1:1, the following text is displayed in the selection pane:
![invalid selection](https://user-images.githubusercontent.com/51806180/60825619-1ca26e80-a17a-11e9-9f32-a90f27c12655.png)

Observation: In the earlier Project(from my first comment of this issue- en_ust_rut_book.zip), for Ruth 1:1- "No selection needed" selection has been made.

**Comment by PhotoNomad0** [comment]
@elsylambert - I double checked and this is a content error in the catalog in the TN tsv.  The original language quote `בָּ​אָ֑רֶץ` is not in Ruth 1:1.  It looks like the program is behaving as it should and the message is correct.  

@benjore - how should we handle content errors going forward?

**Comment by PhotoNomad0** [comment]
Looks like the original text has a word joiner U+2060, but the quote has a zero width space in it U+200B

**Comment by cckozie** [comment]
@benjore  - My 2 cents worth: I think it would be better to display a user friendly alert indicating a problem rather than put it in the translation pane. (A generic user-friendly message, e.g. "There is a problem with the content for this check", along with some more descriptive text if we know it, and the specifics of the check, book, chapter, verse, occurrence )
As part of the alert, request the user to copy and paste the text from the alert into user feedback.
Yes, this could be a new issue.

**Comment by PhotoNomad0** [comment]
@cckozie - that is a very well thought out way of handling the error.  The only thing I would add - since we are already so close to it - is to add a button that would open user feedback with all the important user feedback. @benjore

**Comment by cckozie** [comment]
@PhotoNomad0 @benjore - I had the feedback button with auto fill in there, but deleted it thinking it would be too much for this issue. All in favor of it.

**Comment by benjore** [comment]
That all sounds good to me! Be sure to capture the GL too.

**Comment by PhotoNomad0** [comment]
@benjore - we should probably size this, since it has expanded to become an issue about how to protect tCore and users from bad content.

**Comment by PhotoNomad0** [comment]
@benjore - this needs sizing

**Comment by PhotoNomad0** [comment]
Prototype display:
![Screen Shot 2019-07-23 at 10 19 50 AM](https://user-images.githubusercontent.com/14238574/61719866-bac63500-ad33-11e9-9259-c8632ffcea67.png)

**Comment by PhotoNomad0** [comment]
@benjore - blocked waiting for consensus on behavior and design.

**Comment by PhotoNomad0** [comment]
Revised with more data:
![Screen Shot 2019-07-23 at 1 25 45 PM](https://user-images.githubusercontent.com/14238574/61733307-819abe80-ad4d-11e9-8491-eb34254084aa.png)

**Comment by PhotoNomad0** [comment]
Outstanding questions:
- should we have the "do not show warning again"?
- what should be the behavior on clicking cancel and submit buttons?  Do we skip to next check?

**Comment by benjore** [comment]
- [ ] "Do you want to report this problem so it can be fixed?"
- [ ] Primary Button: "Send Feedback"
  - [ ] When clicked, open Feedback Dialog
    - [ ] When feedback dialog is closed, take them to the next check.
- [ ] Secondary Button: "Ignore"
  - [ ] When clicked, go to next check
- [ ] Remove the "Do not show this warning again." Option.

**Comment by PhotoNomad0** [comment]
Updated screenshot:
![Screen Shot 2019-07-24 at 7 36 33 AM](https://user-images.githubusercontent.com/14238574/61790659-d1c55f80-ade5-11e9-91d9-40cc158d290c.png)

**Comment by elsylambert** [comment]
Verified in 2.0.0 (15ce3cd).
As expected, below Alert message is displayed for the content errors. Send Feedback buttons opens the feedback window and allows user to report the content errors and then moves to the next verse. Ignore Button allows user to the next verse.
![image](https://user-images.githubusercontent.com/51806180/62056764-4df7e280-b1ec-11e9-8228-91309ad392c7.png)

**Comment by elsylambert** [comment]
Feedback window has UI issue on windows:
![image](https://user-images.githubusercontent.com/51806180/62057455-b09dae00-b1ed-11e9-8f16-422eb4152476.png)
Include  Application logs is not displayed.
@cckozie Are you able to reproduce this issue?

**Comment by PhotoNomad0** [comment]
I was able to reproduce on Mac and Windows by artificially decreasing vertical height of window.  Looks like a pre-existing UI layout problem with feedback dialog.

![Screen Shot 2019-07-29 at 11 10 43 AM](https://user-images.githubusercontent.com/14238574/62059555-8817b300-b1f1-11e9-8aac-2e1d8c035d15.png)

**Comment by PhotoNomad0** [comment]
@elsylambert @cckozie @benjore - this should be a new issue for a pre-existing problem.  This is not a QA failure of this issue that is titled "In tN, Code appearing in selection pane when there is no highlight in the scripture pane".  We already had scope creep on this issue, and it looks like it currently meets the agreed upon definition of done.

**Comment by PhotoNomad0** [comment]
PS - you can reproduce the problem on the projects screen by doing send feedback and adding lines of text to feedback.

It looks like it needs to be made scrollable.

**Comment by elsylambert** [comment]
Submitted a new issue for the Feedback window #6285 

For tN, the Alert message works as expected.

**Comment by cckozie** [comment]
Looks good in 2.0.0 (c940391)
I'm not seeing the problem on the Feedback dialog on the Mac.
Needs regression

**Comment by cckozie** [comment]
2.0.0 (c940391)
GL and locale set to Hindi
![image](https://user-images.githubusercontent.com/8171759/62137790-2b2d0300-b2ac-11e9-8735-60020d5364f2.png)

**Comment by PhotoNomad0** [comment]
Missing Crowdin translations for new strings.  Will take a look, but think we are currently blocked by broken strings in Kanada.

**Comment by PhotoNomad0** [comment]
@elsylambert - OK, have translations unblocked.  Here is a screenshot of the updated locale strings in translationCore 2.0.0 (bd37460):

![Screen Shot 2019-07-31 at 8 14 40 AM](https://user-images.githubusercontent.com/14238574/62210931-5165a800-b36b-11e9-8ba3-e7bac4ee6a17.png)

**Comment by cckozie** [comment]
2.0.0 (bd37460) I'm not seeing the invalid locale messages anymore. Everything looks good other than the missing scrollable area.

**Comment by elsylambert** [comment]
Looks good with Hindi Locale in 2.0.0 (bd37460).

Separate issue created for handling Locale issues #6304

**Comment by cckozie** [comment]
In 2.0.0 (8bd068d) the Feedback dialog is hidden behind the problem with content dialog if the next check also has a content problem.
https://git.door43.org/TC_SAVE/en_ust_rut_book
Open 1:16 under Assumed knowledge and click Send Feedback
![image](https://user-images.githubusercontent.com/8171759/62393980-77d54000-b531-11e9-8131-fb3ed149e6e1.png)

**Comment by PhotoNomad0** [comment]
@cckozie - I think I have a fix that will not move to next verse until after you exit the feedback.  This should address the problem you saw.

But I cannot reproduce your problem above on my system using your project since 1:16 check and 2:18 check are both good.  How did you create the failures in the first place?  did you edit the json files?

**Comment by PhotoNomad0** [comment]
We currently keep tN checks up to date in project, so about the only way to break a project would be to break the glQuotes in en tN helps file.

**Comment by PhotoNomad0** [comment]
Modifying `resources⁩ ▸ ⁨en⁩ ▸ ⁨translationHelps⁩ ▸ ⁨translationNotes⁩ ▸ ⁨v16⁩ ▸ ⁨culture⁩ ▸ ⁨groups⁩ ▸ ⁨rut⁩ ▸ figs-explicit.json` for testing.

**Comment by cckozie** [comment]
@PhotoNomad0 - No, I do not believe I modified any files. I checked the timestamps on all of the relevant files and they all are the same as the surrounding ones. I am not seeing English or Hebrew in the available source content updates.
In the project I do not see any highlights in the scripture pane, either in the UHB or ULT on those verses. Does that indicate that they aren't aligned?

**Comment by PhotoNomad0** [comment]
Verified in Version 2.0.0 (9f5e506)

**Comment by PhotoNomad0** [comment]
> @PhotoNomad0 - No, I do not believe I modified any files. I checked the timestamps on all of the relevant files and they all are the same as the surrounding ones. I am not seeing English or Hebrew in the available source content updates.
> In the project I do not see any highlights in the scripture pane, either in the UHB or ULT on those verses. Does that indicate that they aren't aligned?

@cckozie - don't worry about it - came up with an alternate way to reproduce the bug you saw.  With latest fix will not change to next verse until after feedback is exited.

**Comment by cckozie** [comment]
Works correctly in 2.0.0 (61fec46) 
Needs regression

**Comment by elsylambert** [comment]
Verified in 2.0.0 (92b46cf). Working fine.
